src/lib/storage/pos_list.hpp

```diff
@@ -64,10 +64,12 @@ struct PosList final : private pmr_vector<RowID> {
           [&]() {
             if (size() == 0) return true;
             const auto& common_chunk_id = (*this)[0].chunk_id;
-            return std::all_of(cbegin(), cend(),
-                               [&](const auto& row_id) { return row_id.chunk_id == common_chunk_id; });
+            if (common_chunk_id == INVALID_CHUNK_ID) return false;
+            return std::all_of(cbegin(), cend(), [&](const auto& row_id) {
```

Comment:
What is the problem with NULL values? Why don't we want them?

Reply:
Above, we say

> If all entries in the PosList shares a single ChunkID, it makes sense to explicitly give this guarantee in order to enable some optimizations.

Does changing it to

> If we know that all entries in the PosList share a single ChunkID, we can optimize the indirection by retrieving the respective chunk once and using only the ChunkOffsets to access values. As the consumer will likely call `table->get_chunk(common_chunk_id)`, we require that the common chunk id is valid, i.e., the PosList contains no NULL values.

help?

Reply:
That does indeed help. Will resolve this issue.

Just for my understanding: is that something that operators actually require right now or is it a preparation for PosList optimization (e.g., storing no chunk_id for every row)?
I'd guess operators wouldn't mind NULL values even when using optimizations for such single-chunk-ref chunks?!??

Reply:
We are not talking about NULL values in, e.g., the DictionarySegment being referenced, but about NULL values in the PosList itself. Currently this does not happen, because no operator that guarantees the single chunk property sets such NULL values, but when I played around with it for other operators, I ran into this issue. To avoid other people setting the property falsely, which may break seemingly unrelated downstream operators, I added this check.